ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+        with open(parmfile) as f3:
+            parmvars = f3.read().splitlines()
+
+        nvarsall=len(parmvars)
+        nvars=0
+        nvpairs=0
+        vartype=np.zeros(shape=(nvarsall),dtype=list)
+        for nva in range(0,nvarsall):
+            if ',' in parmvars[nva]:
+                vartype[nva]='vector'
+                nvpairs=nvpairs+1
+            else:
+                vartype[nva]='scalar'
+                nvars=nvars+1
+
+        parmvars_s=np.array(parmvars)[np.argwhere(vartype=='scalar')]
+        parmvars_v=np.array(parmvars)[np.argwhere(vartype=='vector')]
+
+        varlist=np.zeros(shape=(nvars,4),dtype=list)
+        for nv in range(0,nvars):
+            nvarparm1=np.char.split(parmvars_s[nv],':')
+            varlist[nv,:]=nvarparm1[0]           
+
+        veclist=np.zeros(shape=(nvpairs,4,2),dtype=list)
+        for nv in range(0,nvpairs):
+            nvectparm1=np.char.split(parmvars_v[nv],':')
+            nvectparm2=nvectparm1[0]      
+            for nvp in range(0,4):
+                veclist[nv,nvp,:]=nvectparm2[nvp].split(',')
+
+    #----------------------------------------------------------------------
+    # make a list of the directories and files from the run 
+    #----------------------------------------------------------------------
+
+    #specify input file name and input directory
+    print('Extracting Input Data')
+    ocnf=nc.Dataset(infiles1[0])  #Read using nc
+    
+
+    #Get a list of available variables from netcdf file
+    ncvarlist=ocnf.variables.keys()  #Get list of variables using nc
+
+    #Check which model is to be post-processed 
+    #by searching for a key variable specific to that model
+    if 'SST' in ncvarlist:
+        model='MOM'
+    elif 'Tsfc_d' in ncvarlist or 'Tsfc_h' in ncvarlist:
+        model='CICE'
+        if 'Tsfc_d' in ncvarlist:
+            samplevar='Tsfc_d'
+        if 'Tsfc_h' in ncvarlist:
+            samplevar='Tsfc_h' 
+    else:
+        print('Product not supported. Exiting program...')
+        exit()
+    print('Post-processing '+str(model))
+
+    #Extract Source Grid coordinates
+    if model == 'MOM':
+        dimxh=ocnf.dimensions['xh'].size
+        dimyh=ocnf.dimensions['yh'].size
+    if model == 'CICE':
+        dimxh=ocnf.dimensions['ni'].size
+        dimyh=ocnf.dimensions['nj'].size
+
+    print('Getting rotation angles and vertical grid')
+    #get the rotation angles and vertical grid from the first file
+    #two different name were used for the angles, either sinrot,cosrot 
+    #or sin_rot,cos_rot
+    if model == 'MOM':
+        if 'sin_rot' in ncvarlist:
+            sinrot=ocnf['sin_rot']
+        else:
+            sinrot=ocnf['sinrot']
+        if 'cos_rot' in ncvarlist:
+            cosrot=ocnf['cos_rot']
+        else:
+            cosrot=ocnf['cosrot']
+
+        z_l=ocnf['z_l']
+        z_i=ocnf['z_i']
+        nlevs=len(z_l)
+
+    if model == 'CICE':
+        angleT=ocnf['ANGLET'] 
```

Comment:
Handle `MOM` in `MOM` related methods and `CICE` likewise.

Reply:
This is not necessary. Both MOM6 and CICE are on the same tri-polar grid for the UFS. Further, there is no rotation necessary for CICE since the variables are all Arakawa-C mass points.

Reply:
Can we confirm with @DeniseWorthen who wrote the original NCL scripts.  While I don't remember specific details, I do know that Denise is an expert on the CICE grid and carefully confirmed every variable was being interpolated the correct way and I'm assuming Eric is following what was done in the original NCL scripts here. Although maybe somethings changed since then. 

PS Thank you @EricSinsky-NOAA for this PR.  I know this has been a long standing need (https://github.com/NOAA-EMC/global-workflow/issues/923 was created a year ago) and it's greatly appreciated.

Reply:
Thanks @JessicaMeixner-NOAA. The vector rotation is being performed the same way as in Denise's NCL code. It is possible, however, that something has changed since then (as you pointed out).